pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -15,8 +15,9 @@ type InfinispanAuthInfo struct {
 
 // InfinispanSecurity info for the user application connection
 type InfinispanSecurity struct {
-	EndpointSecretName string              `json:"endpointSecretName,optional,omitempty"`
-	EndpointEncryption *EndpointEncryption `json:"endpointEncryption,optional,omitempty"`
+	EndpointAuthentication *bool               `json:"endpointAuthentication,optional,omitempty"`
```

Comment:
It would be nice to have an enum for enable, disable or set another authentication type as described in #696 
This is will give flexibility with the future authentication options

Reply:
What other type of authentication options do you envisage being required that could be satisfied with a single field?

Bear in mind that once we move to the next CR version we will re-work `InfinispanSecuriy` anyway so that it's more flexible. For example, this was a potential option discussed on zulip:

```yaml
spec:
  security:
    authorization: ...
    authentication:
      enabled: true | false
      secretName: ...
    endpoints:
      encryption: ...
```

Reply:
It can be an auth type like None, for disable and anything else for container embedded ocp/k8s.
Let's keep current proposal and change it in the next CRD version.

Reply:
Sounds good. I have removed `InfinispanAuthInfo` as it's a redundant struct.